mips/cbus.c

```diff
@@ -2,20 +2,20 @@
 #include <bus.h>
 
 static uint8_t cbus_read_1(resource_t *handle, unsigned offset) {
-  intptr_t addr = handle->r_addr + offset * sizeof(uint64_t);
-  return *(volatile uint8_t *)(MIPS_PHYS_TO_KSEG1(addr));
+  intptr_t addr = handle->r_start + offset * sizeof(uint64_t);
+  return *(volatile uint8_t *)MIPS_PHYS_TO_KSEG1(addr);
 }
 
 static void cbus_write_1(resource_t *handle, unsigned offset, uint8_t value) {
-  intptr_t addr = handle->r_addr + offset * sizeof(uint64_t);
-  *(volatile uint8_t *)(MIPS_PHYS_TO_KSEG1(addr)) = value;
+  intptr_t addr = handle->r_start + offset * sizeof(uint64_t);
+  *(volatile uint8_t *)MIPS_PHYS_TO_KSEG1(addr) = value;
 }
 
 static bus_space_t cbus_bus_space = {
   .read_1 = cbus_read_1, .write_1 = cbus_write_1,
 };
 
 resource_t cbus_uart[1] = {{.r_bus_space = &cbus_bus_space,
-                            .r_addr = MALTA_CBUS_UART,
-                            .r_start = 0,
-                            .r_end = 7}};
+                            .r_type = RT_MEMORY,
```

Comment:
I initially thought the same, but... Firstly, MIPS architecture has no notion of hardware I/O ports, as known from PC platform. Secondly, CBUS-UART is connected directly to the CPU by common address/data bus. Although PCI I/O space is reachable through memory as well, it's clear (for me?) that all resources mapped there should by marked with `RT_IOPORTS`.